akka-projection-eventsourced/src/main/scala/akka/projection/eventsourced/scaladsl/EventSourcedProvider.scala

```diff
@@ -273,6 +310,271 @@ object EventSourcedProvider {
 
   }
 
+  private class ReplayableEventsBySlicesSourceProvider[Event](
+      override val system: ActorSystem[_],
+      query: EventsBySliceQuery with CurrentEventsByPersistenceIdTypedQuery,
+      entityType: String,
+      override val minSlice: Int,
+      override val maxSlice: Int,
+      adjustStartOffset: Option[Offset] => Future[Option[Offset]])
+      extends EventsBySlicesSourceProvider[Event](system, query, entityType, minSlice, maxSlice, adjustStartOffset)
+      with CanTriggerReplay
+      with HasGapFillerActor {
+
+    /** INTERNAL API */
+    @InternalApi
+    override private[scaladsl] val gapFillerRef = new AtomicReference()
+
+    override val entityTypeSliceRange = s"$entityType-$minSlice-$maxSlice"
+    override def byPersistenceIdQuery: CurrentEventsByPersistenceIdTypedQuery = query
+
+    override private[akka] def triggerReplay(persistenceId: String, fromSeqNr: Long, triggeredBySeqNr: Long): Unit =
+      fill(persistenceId, fromSeqNr, triggeredBySeqNr)
```

Comment:
So this would trigger a replay for that pid in any source created from this provider instance. That might be fine because we generally show creating a provider instance specifically where it is used, so not really expected to be shared.

I guess that explains why you made the replay logic an actor that can be shared and needs to manage it's lifecycle. Quite inconvenient that it cannot be about a specific running query when that is what we'd really want.